### ISSUE TITLE
bossfetch not very informative when $BOSS_LOCAL_ROOT is not set

### ISSUE THREAD

**Issue by dmargala** [dkirkby/bossdata#5]
As the snippet below shows, an error occurs when I try to run `bossfetch` without setting `$BOSS_LOCAL_ROOT`. However, the error message is not very helpful at understanding the root cause. I had to change my environment variable name from `$BOSS_DATA_ROOT` to `$BOSS_LOCAL_ROOT`.

```
$ bin/bossfetch --verbose targets.txt 
Fetching 160473 observations...
Skipping file with download error: /sas/dr12/boss/spectro/redux/v5_7_0/spectra/lite/6173/spec-6173-56238-0528.fits.  |
Traceback (most recent call last):
  File "bin/bossfetch", line 76, in <module>
    main()
  File "bin/bossfetch", line 61, in main
    num_bytes += os.path.getsize(local_path)
UnboundLocalError: local variable 'local_path' referenced before assignment
```

**Comment by dkirkby** [comment]
I made the checking of env vars more uniform and this now fails with:

```
% unset BOSS_LOCAL_ROOT
% bin/bossfetch missing.dat
No local root specified: try setting $BOSS_LOCAL_ROOT.
```

Behind the scenes, the following two commands (which most scripts start with) now check that all 4 env vars are set and return a `ValueError` with a consistent error message:

```
finder = bossdata.path.Finder()  # requires BOSS_SAS_ROOT and BOSS_REDUX_VERSION
mirror = bossdata.remote.Manager()  # requires BOSS_DATA_URL and BOSS_LOCAL_ROOT
```

**Comment by dkirkby** [comment]
Fixed in 0.2
